### PR TITLE
app_store: new API and split backend

### PR DESCRIPTION
**Pull request by barraguda**
WIP

### CODE REVIEW

File: kinode/packages/app_store/api/app_store:sys-v0.wit [hunk 1]
```diff
@@ -1,20 +1,107 @@
...
+        add-download(add-download-request),
+    }
+
+    record download-request {
+        package-id: package-id,
+        download-from: option<string>, // note, might be too much implicit complexity,
+                                       // but: download-from received locally, is an instruction where to download-from
+                                       // and if received from remote, is an instruction where to send, where the "download-from" came from.
+        desired-version-hash: string,
+    }
```
Comment (dr-frmr) [inline kinode/packages/app_store/api/app_store:sys-v0.wit]: refactor into remote-download-request with no `download-from` field, and local-download-request with `download-from: string`

File: kinode/packages/app_store/api/app_store:sys-v0.wit [hunk 2]
```diff
@@ -1,20 +1,107 @@
...
+
+    record download-response {
+        success: bool,
+        error: option<string>,
```
Comment (dr-frmr) [inline kinode/packages/app_store/api/app_store:sys-v0.wit]: change `error` into an enum type that lists all possible download errors, change `download-response` into a Result at the top level, so like `Result<(), DownloadError>`

File: kinode/packages/app_store/api/app_store:sys-v0.wit [hunk 3]
```diff
@@ -1,20 +1,107 @@
...
+
+    record size-update {
+        package-id: package-id,
+        size: u64,
+    }
+}
+
+interface chain {
+    //
+    // on-chain API as presented by chain:app_store:sys-v0
```
Comment (dr-frmr) [inline kinode/packages/app_store/api/app_store:sys-v0.wit:111]: style preference: `on-chain` should be `onchain` always

File: kinode/packages/app_store/api/app_store:sys-v0.wit [hunk 4]
```diff
@@ -1,20 +1,107 @@
...
+    //
+
+    use standard.{package-id};
+
+    variant chains {
```
Comment (dr-frmr) [inline kinode/packages/app_store/api/app_store:sys-v0.wit]: why is this called `chains`? can it be labeled better as a Request enum?

File: kinode/packages/app_store/api/app_store:sys-v0.wit [hunk 5]
```diff
@@ -26,79 +113,79 @@ interface main {
...
+    use standard.{package-id};
+    use chain.{on-chain-metadata};
+
+
+    // general error type used by both chain: and downloads: 
+    // todo: add variant for types to match on
+
+    record error {
+        reason: string,
     }
```
Comment (dr-frmr) [inline kinode/packages/app_store/api/app_store:sys-v0.wit]: yup, turn into an enum

File: kinode/packages/app_store/ui/vite.config.ts.timestamp-1723130861144-28181aac42611.mjs
Comment (dr-frmr) [inline kinode/packages/app_store/ui/vite.config.ts.timestamp-1723130861144-28181aac42611.mjs:1]: make sure to delete this from final PR

File: kinode/packages/app_store/install/src/lib.rs [hunk 6]
```diff
@@ -34,13 +34,13 @@ fn init(our: Address) {
...
-                    },
-                ))
-                .unwrap(),
+                b"temp", // serde_json::to_vec(&LocalRequest::Install(
+                        //     crate::kinode::process::main::PackageId {
+                        //         package_name: package_id.package_name.clone(),
+                        //         publisher_node: package_id.publisher_node.clone(),
+                        //     },
+                        // ))
+                        // .unwrap(),
```
Comment (dr-frmr) [inline kinode/packages/app_store/install/src/lib.rs]: how do you want to implement this? add version hash to the command like for download?

Comment (barraguda) [reply kinode/packages/app_store/install/src/lib.rs]: yep, that's the plan

**Comment by barraguda** [comment]
merging into develop now, 

I'll tweak the UI slightly with a separate PR later, hiding some of the more advanced features from users by default to not overload the pages.

**Pull request merged**